src/Handler/Table/Import/ImportTableFromTableHandler.php

```diff
@@ -219,32 +220,52 @@ private function import(
             return [null, $importState->getResult()];
         }
 
-        $stagingTable = $this->createStageTable($destinationDefinition, $sourceMapping, $bqClient);
+        /** @var TableImportFromTableCommand\SourceTableMapping\ColumnMapping[] $mappings */
+        $mappings = iterator_to_array($sourceMapping->getColumnMappings()->getIterator());
+        $stagingTable = StageTableDefinitionFactory::createStagingTableDefinitionWithMapping(
+            $destinationDefinition,
+            $mappings,
+        );
+        // if column names are identical, we can use CREATE TABLE COPY (CopyImportFromTableToTable)
+        // if column names are not identical, we have to use INSERT INTO (ToStageImporter)
+        $isColumnNameIdentical = true;
+        foreach ($mappings as $columnMapping) {
+            if ($columnMapping->getSourceColumnName() === $columnMapping->getDestinationColumnName()) {
+                continue;
+            }
+            $isColumnNameIdentical = false;
+        }
         // load to staging table
-        $toStageImporter = new ToStageImporter($bqClient);
+        if ($isColumnNameIdentical) {
+            $toStageImporter = new CopyImportFromTableToTable($bqClient);
+        } else {
+            $stagingTable = $this->createStageTable($destinationDefinition, $sourceMapping, $bqClient);
+            // load to staging table
+            $toStageImporter = new ToStageImporter($bqClient);
+        }
         try {
             $importState = $toStageImporter->importToStagingTable(
                 $source,
                 $stagingTable,
                 $importOptions,
             );
+            // import data to destination
+            $toFinalTableImporter = new FullImporter($bqClient);
+            if ($importOptions->isIncremental()) {
+                $toFinalTableImporter = new IncrementalImporter($bqClient);
+            }
+            $importResult = $toFinalTableImporter->importToTable(
```

Comment:
Úplně nevím, jak tohle má fungovat.

Ale přijde mi to divné. Nejdřív importuješ pomocí `toStageImporter` který je potencionálně `CopyImportFromTableToTable` takže tam už data naleje.

Ale pak ještě nalejváš data přes `FullImporter`

Bohužel tady nemám dostatečnou znalost BQ …

Reply:
Stage importer importuje do dočasné stage tabulky, Full nebo Incremental pak importuje ze stage tabulky do final tabulky.

Reply:
OK, v tom případě to vypadá dobře :-)